### DISCUSSION TITLE
Streaming responses with APIGatewayRestResolver

### DISCUSSION THREAD

**Discussion by myslak71** [aws-powertools/powertools-lambda-python#5097] [Q&A]
Is there a way to return streaming response within async endpoint like so:
```
@app.get("/ping")
async def ping() -> StreamingResponse:
    s3 = S3Object(
        bucket="bucket_name",
        key="object-key",
    )
    return StreamingResponse(s3, ...)
```

**Comment by leandrodamascena** [comment]
Hey @myslak71! I completely missed this discussion, really sorry!

Our Event Handler Resolvers does not support async out of the box, but we have customers using solutions to use this. Please check this thread and see if it helps: https://github.com/aws-powertools/powertools-lambda-python/issues/3934#issuecomment-2241724333

**Comment by leandrodamascena** [comment]
I'm closing this discussion as resolved. Please reopen if you need.
